crates/turbopack-node/js/src/ipc/evaluate.ts

```diff
@@ -13,27 +12,28 @@ type IpcOutgoingMessage = {
 
 const ipc = IPC as Ipc<IpcIncomingMessage, IpcOutgoingMessage>;
 
-export const run = (getValue: () => any) => {
-  (async () => {
-    while (true) {
-      const msg = await ipc.recv();
+export const run = async (getValue: (...deserializedArgs: any[]) => any) => {
+  while (true) {
+    const msg = await ipc.recv();
 
-      switch (msg.type) {
-        case "evaluate": {
-          const value = await getValue();
-          await ipc.send({
-            type: "jsonValue",
-            data: JSON.stringify(value),
+    switch (msg.type) {
+      case "evaluate": {
+        const value = await Promise.resolve()
+          .then(() => getValue(...msg.args))
+          .catch((err: Error) => {
```

Comment:
Nit: This could just be a try-catch and be less confusing:
```js
try {
  const value = await getValue(...msg.args);
  await ipc.send({
    type: "jsonValue",
    data: JSON.stringify(value),
  });
} catch (e) {
  await ipc.sendError(e as Error);
}
```